### PR TITLE
Set impl

### PR DESCRIPTION
**Pull request by GhostInAMachine**

### CODE REVIEW

File: test/SetProxyTest.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        }).then(function () {
+            return setInstance.containsAll([3, 4]).then(function (contains) {
+                expect(contains).to.be.false;
+            });
+        });
+    });
+
+
+    it("is empty", function () {
+        setInstance.isEmpty().then(function (empty) {
```
Comment (mustafaiman) [inline test/SetProxyTest.js]: Make sure to `return` promises from tests. This test always passes regardless of return value.

File: test/SetProxyTest.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            expect(empty).to.be.true;
+            return setInstance.add(1);
+        }).then(function () {
+            setInstance.isEmpty().then(function (empty) {
```
Comment (mustafaiman) [inline test/SetProxyTest.js]: `return` here too.

Comment (GhostInAMachine) [inline test/SetProxyTest.js]: Why? There is no promise chaining there.

Comment (GhostInAMachine) [inline test/SetProxyTest.js]: Nevermind, I see

Comment (mustafaiman) [inline test/SetProxyTest.js]: This is related to mocha. This test always passes no matter what return values are. Try it yourself.

**Comment by mustafaiman** [comment]
See: https://hazelcast-l337.ci.cloudbees.com/job/NodeJS-pr-builder/38/console

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
verify

**Comment by mustafaiman** [comment]
👍

**Pull request merged**